libraries/AP_Logger/LogStructure.h

```diff
@@ -136,7 +136,13 @@ const struct MultiplierStructure log_Multipliers[] = {
 #include <AP_AIS/LogStructure.h>
 
 // structure used to define logging format
+// It is packed to save flash space; however, this causes problems when building
+// the SITL on an Apple M1 CPU so we do not pack it when compiling SITL
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
```

Comment:
make it PACKED only on HAL_CHIBIOS
that will ensure all the linux ports are correct, and flash space is minimized on stm32

Reply:
done